stdlib/date/date.rbs

```diff
@@ -0,0 +1,1057 @@
+#      #=> #<Date: 2001-02-03 ...>
+#
+# All date objects are immutable; hence cannot modify themselves.
+#
+# The concept of a date object can be represented as a tuple of the day count,
+# the offset and the day of calendar reform.
+#
+# The day count denotes the absolute position of a temporal dimension. The
+# offset is relative adjustment, which determines decoded local time with the
+# day count. The day of calendar reform denotes the start day of the new style.
+# The old style of the West is the Julian calendar which was adopted by Caesar.
+# The new style is the Gregorian calendar, which is the current civil calendar
+# of many countries.
+#
+# The day count is virtually the astronomical Julian day number. The offset in
+# this class is usually zero, and cannot be specified directly.
+#
+# A Date object can be created with an optional argument, the day of calendar
+# reform as a Julian day number, which should be 2298874 to 2426355 or
+# negative/positive infinity. The default value is `Date::ITALY`
+# (2299161=1582-10-15). See also sample/cal.rb.
+#
+#     $ ruby sample/cal.rb -c it 10 1582
+#         October 1582
+#      S  M Tu  W Th  F  S
+#         1  2  3  4 15 16
+#     17 18 19 20 21 22 23
+#     24 25 26 27 28 29 30
+#     31
+#
+#     $ ruby sample/cal.rb -c gb  9 1752
+#        September 1752
+#      S  M Tu  W Th  F  S
+#            1  2 14 15 16
+#     17 18 19 20 21 22 23
+#     24 25 26 27 28 29 30
+#
+# A Date object has various methods. See each reference.
+#
+#     d = Date.parse('3rd Feb 2001')
+#                                  #=> #<Date: 2001-02-03 ...>
+#     d.year                       #=> 2001
+#     d.mon                        #=> 2
+#     d.mday                       #=> 3
+#     d.wday                       #=> 6
+#     d += 1                       #=> #<Date: 2001-02-04 ...>
+#     d.strftime('%a %d %b %Y')    #=> "Sun 04 Feb 2001"
+#
+class Date
+  # Creates a date object denoting the given calendar date.
+  #
+  # In this class, BCE years are counted astronomically.  Thus, the year before
+  # the year 1 is the year zero, and the year preceding the year zero is the year
+  # -1.  The month and the day of month should be a negative or a positive number
+  # (as a relative month/day from the end of year/month when negative).  They
+  # should not be zero.
+  #
+  # The last argument should be a Julian day number which denotes the day of
+  # calendar reform.  Date::ITALY (2299161=1582-10-15), Date::ENGLAND
+  # (2361222=1752-09-14), Date::GREGORIAN (the proleptic Gregorian calendar) and
+  # Date::JULIAN (the proleptic Julian calendar) can be specified as a day of
+  # calendar reform.
+  #
+  #     Date.new(2001)            #=> #<Date: 2001-01-01 ...>
+  #     Date.new(2001,2,3)        #=> #<Date: 2001-02-03 ...>
+  #     Date.new(2001,2,-1)       #=> #<Date: 2001-02-28 ...>
+  #
+  # See also ::jd.
+  #
+  def self.new: (?Integer year, ?Integer month, ?Integer mday, ?Integer start) -> Date
```

Comment:
Could you delete the `.new` method definition? It's automatically generated from `#initialize` type. (Same for `DateTime.new` too.)

(And the return type of `initialize` can be `void`.)

Reply:
Oh, it's convenient. I understand and deleted it (date and datetime).